src/plugins/credential_management/public/components/common/components/create_credential_form/create_credential_form.tsx

```diff
@@ -71,6 +72,16 @@ export class CredentialForm extends React.Component<CredentialFormProps, Credent
     };
   }
 
+  componentDidUpdate(
+    prevProps: Readonly<CredentialFormProps>,
+    prevState: Readonly<CredentialFormState>,
+    snapshot?: any
+  ) {
+    if (prevProps.callSubmit !== this.props.callSubmit) {
```

Comment:
Did't find where where callSubmit will be updated, aka will callSubmit ever change?

Reply:
Yes .. It is chanigng from edit_datasource_form.tsx